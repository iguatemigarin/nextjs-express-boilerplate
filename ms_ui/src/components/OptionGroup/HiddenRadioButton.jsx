import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

// TODO: Refactor with "as" polymorphic prop from Styled Components 4
// https://snook.ca/archives/html_and_css/hiding-content-for-accessibility
const Input = styled('input')`
  position: absolute !important;
  height: 1px; width: 1px;
  overflow: hidden;
  clip: rect(1px, 1px, 1px, 1px);
`;

const HiddenRadioButton = ({ id, name, value }) => (
  <Input type="radio" id={id} name={name} value={value} />
);

HiddenRadioButton.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.node.isRequired,
};

export default HiddenRadioButton;
