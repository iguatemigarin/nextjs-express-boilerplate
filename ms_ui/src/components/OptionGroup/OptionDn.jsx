import PropTypes from 'prop-types';
import React, { memo } from 'react';
import styled, { css } from 'styled-components';
import HiddenRadioButton from './HiddenRadioButton';

const Label = styled('label')`
  flex: 1;
  cursor: pointer;
  border-top: 2px solid ${props => props.theme.colors.primary};
  border-right: 2px solid ${props => props.theme.colors.primary};
  border-bottom: 2px solid ${props => props.theme.colors.primary};
  background-color: ${props => props.theme.colors.white};
  padding: 11px 0;
  text-align: center;
  font: ${props => props.theme.fontScale.small} ${props => props.theme.fontFamilies.regular};

  :hover, :focus {
    background-color: ${props => props.theme.colors.primary_15};
  }

  :first-child {
    border-left: 2px solid ${props => props.theme.colors.primary};
    border-top-left-radius: ${props => props.theme.borderRadius.normal};
    border-bottom-left-radius: ${props => props.theme.borderRadius.normal};
  }
  
  :last-child {
    border-top-right-radius: ${props => props.theme.borderRadius.normal};
    border-bottom-right-radius: ${props => props.theme.borderRadius.normal};
  }
  
  ${props => props.checked && css`
    background-color: ${props.theme.colors.primary_30};
    font-family: ${props.theme.fontFamilies.demi};
  `}
`;

const OptionDn = ({
  name,
  label,
  value,
  checked,
  onChecked,
}) => {
  const domId = [name, value].join('__');

  return (
    <Label htmlFor={domId} checked={checked} onChange={() => onChecked({ label, value })}>
      <HiddenRadioButton id={domId} name={name} value={value} />
      {label}
    </Label>
  );
};

OptionDn.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  value: PropTypes.node.isRequired,
  checked: PropTypes.bool.isRequired,
  onChecked: PropTypes.func.isRequired,
};

export default memo(OptionDn);
