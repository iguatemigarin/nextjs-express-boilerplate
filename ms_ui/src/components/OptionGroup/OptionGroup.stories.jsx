import { object } from '@storybook/addon-knobs';
import React from 'react';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';
import { msTheme } from '../../themes/index';
import OptionGroup, { OptionGroupComponent } from '.';

// TODO: Move Storybook-only wrapper components somewhere to be reused
const Wrapper = styled.div`
  ${({ theme }) => `
    padding: 40px 0;
    background: ${theme.tonals.t30};
  `}
`;

// TODO: Replace with a CSS grid width, once available
const Content = styled.div`
  margin: 0 auto;
  max-width: 455px;
`;

const options = [
  { value: 0, label: 'Mon' },
  { value: 1, label: 'Tue' },
  { value: 2, label: 'Wed' },
  { value: 3, label: 'Thu' },
];

storiesOf('OptionGroup', module)
  .addDecorator(story => (
    <Wrapper theme={msTheme}>
      <Content>
        {story()}
      </Content>
    </Wrapper>
  ))
  .addParameters({
    info: {
      text: 'Button group component',
      propTables: [OptionGroupComponent],
      propTablesExclude: [Wrapper, Content, OptionGroup],
    },
  })
  .add('day picker', () => (
    <OptionGroup name="deliveryDay" options={object('options', options)} />
  ));
