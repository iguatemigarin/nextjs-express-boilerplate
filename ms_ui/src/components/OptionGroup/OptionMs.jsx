import PropTypes from 'prop-types';
import React, { memo } from 'react';
import styled, { css } from 'styled-components';
import HiddenRadioButton from './HiddenRadioButton';

const Label = styled('label')`
  background-color: ${props => props.theme.colors.white};
  border: 1px solid ${props => props.theme.colors.pepper_black_35};
  border-right: none;
  border-top: 6px solid ${props => props.theme.colors.pepper_black_15};
  box-shadow: 0 -1px 0 ${props => props.theme.colors.pepper_black_35};
  cursor: pointer;
  flex: 1;
  font: ${props => props.theme.fontScale.small} ${props => props.theme.fontFamilies.regular}
  margin: 0 0 25px;
  padding: 15px 0;
  text-align: center;

  ${props => props.checked && css`
    border-top-color: ${props.theme.colors.primary};
    font-family: ${props.theme.fontFamilies.demi};
  `}

  :first-child {
    border-top-left-radius: 6px;
    border-bottom-left-radius: 6px;
  }

  :last-child {
    border-right: 1px solid ${props => props.theme.colors.pepper_black_35};
    border-top-right-radius: 6px;
    border-bottom-right-radius: 6px;
  }
`;

const Option = ({
  name,
  label,
  value,
  checked,
  onChecked,
}) => {
  const domId = [name, value].join('__');

  return (
    <Label htmlFor={domId} checked={checked} onChange={() => onChecked({ label, value })}>
      <HiddenRadioButton id={domId} name={name} value={value} />
      {label}
    </Label>
  );
};

Option.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  value: PropTypes.node.isRequired,
  checked: PropTypes.bool.isRequired,
  onChecked: PropTypes.func.isRequired,
};

export default memo(Option);
