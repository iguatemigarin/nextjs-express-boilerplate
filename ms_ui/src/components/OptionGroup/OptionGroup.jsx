import PropTypes from 'prop-types';
import React, { useState } from 'react';
import styled, { withTheme } from 'styled-components';
import OptionMs from './OptionMs';
import OptionDn from './OptionDn';

const Wrapper = styled.div`
  display: flex;
`;

function selectDefaultOption(options = []) {
  return options.length > 0 ? options[0] : null;
}

function OptionGroup({
  name,
  options,
  onSelect,
  theme,
}) {
  const [selected, setSelected] = useState(selectDefaultOption(options));

  function onChecked(option) {
    setSelected(option);
    onSelect(option);
  }

  const Option = theme.brand === 'ms' ? OptionMs : OptionDn;

  return (
    <Wrapper>
      {
        options.map((option) => {
          const checked = selected.value === option.value;
          return (
            <Option
              key={option.value}
              name={name}
              checked={checked}
              onChecked={onChecked}
              {...option}
            />
          );
        })
      }
    </Wrapper>
  );
}

OptionGroup.propTypes = {
  name: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(PropTypes.shape({
    key: PropTypes.string,
    value: PropTypes.node.isRequired,
  })),
  onSelect: PropTypes.func,
  theme: PropTypes.shape({ brand: PropTypes.string }).isRequired,
};

OptionGroup.defaultProps = {
  options: [],
  onSelect: () => {},
};

export const OptionGroupComponent = OptionGroup;
export default withTheme(OptionGroup);
