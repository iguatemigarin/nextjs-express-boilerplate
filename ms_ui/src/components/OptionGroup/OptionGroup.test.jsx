import OptionMs from './OptionMs';
import OptionDn from './OptionDn';
import OptionGroup from './OptionGroup';

let onSelectMock;
const exampleGroup = mockFn => (
  <OptionGroup
    name="deliveryDay"
    options={[
      { value: 0, label: 'Mon' },
      { value: 1, label: 'Tue' },
      { value: 2, label: 'Wed' },
      { value: 3, label: 'Thu' },
    ]}
    onSelect={mockFn}
  />
);

describe('OptionGroup', () => {
  beforeEach(() => {
    onSelectMock = jest.fn();
  });

  it('renders correctly with ms theme', () => {
    expect(toJSON(mountWithTheme(exampleGroup(onSelectMock), msTheme))).toMatchSnapshot();
  });

  it('renders correctly with dn theme', () => {
    expect(toJSON(mountWithTheme(exampleGroup(onSelectMock), dnTheme))).toMatchSnapshot();
  });

  it('renders a radio button per option', () => {
    const wrapper = mountWithTheme(exampleGroup(onSelectMock));
    expect(wrapper.find('input[type="radio"]')).toHaveLength(4);
  });

  it('renders a radio button per option with dn theme', () => {
    const wrapper = mountWithTheme(exampleGroup(onSelectMock), dnTheme);
    expect(wrapper.find('input[type="radio"]')).toHaveLength(4);
  });

  // FIXME: I don't really like how this test works. IMO we're testing internal structure with the
  // selection of an Option component. Ideally, we would properly mount the option group and select
  // the plain HTML input and check if it's checked. But this does not work with React and they say
  // it's intentional...
  // See here https://github.com/facebook/react/issues/4950
  it('selects first option as selected default', () => {
    const wrapper = mountWithTheme(exampleGroup(onSelectMock));
    expect(wrapper.find(OptionMs).filter({ value: 0 }).prop('checked')).toBe(true);
  });

  it('selects first option as selected default with dn theme', () => {
    const wrapper = mountWithTheme(exampleGroup(onSelectMock), dnTheme);
    expect(wrapper.find(OptionDn).filter({ value: 0 }).prop('checked')).toBe(true);
  });

  describe('user interaction', () => {
    let optionGroup;

    beforeEach(() => {
      optionGroup = mountWithTheme(exampleGroup(onSelectMock));
    });

    it('changes the selected item on click of the option', () => {
      expect(optionGroup.find(OptionMs).filter({ value: 2 }).prop('checked')).toBeFalsy();
      optionGroup.find(OptionMs).filter({ value: 2 }).simulate('change');
      expect(optionGroup.find(OptionMs).filter({ value: 2 }).prop('checked')).toBeTruthy();
    });

    it('calls the event handler on change', () => {
      optionGroup.find(OptionMs).filter({ value: 2 }).simulate('change');
      expect(onSelectMock).toBeCalledWith({ value: 2, label: 'Wed' });
    });
  });

  describe('user interaction with dn theme', () => {
    let optionGroup;

    beforeEach(() => {
      optionGroup = mountWithTheme(exampleGroup(onSelectMock), dnTheme);
    });

    it('changes the selected item on click of the option', () => {
      expect(optionGroup.find(OptionDn).filter({ value: 2 }).prop('checked')).toBeFalsy();
      optionGroup.find(OptionDn).filter({ value: 2 }).simulate('change');
      expect(optionGroup.find(OptionDn).filter({ value: 2 }).prop('checked')).toBeTruthy();
    });

    it('calls the event handler on change', () => {
      optionGroup.find(OptionDn).filter({ value: 2 }).simulate('change');
      expect(onSelectMock).toBeCalledWith({ value: 2, label: 'Wed' });
    });
  });
});
