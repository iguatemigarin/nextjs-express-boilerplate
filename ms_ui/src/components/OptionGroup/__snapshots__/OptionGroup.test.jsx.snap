// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionGroup renders correctly with dn theme 1`] = `
.c2 {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: pointer;
  border-top: 2px solid #21bbb1;
  border-right: 2px solid #21bbb1;
  border-bottom: 2px solid #21bbb1;
  background-color: #ffffff;
  padding: 11px 0;
  text-align: center;
  font: 14px/1.5 "Galano Grotesque Regular",sans-serif;
  background-color: #bceae7;
  font-family: "Galano Grotesque Medium",sans-serif;
}

.c1:hover,
.c1:focus {
  background-color: #def5f4;
}

.c1:first-child {
  border-left: 2px solid #21bbb1;
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
}

.c1:last-child {
  border-top-right-radius: 8px;
  border-bottom-right-radius: 8px;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: pointer;
  border-top: 2px solid #21bbb1;
  border-right: 2px solid #21bbb1;
  border-bottom: 2px solid #21bbb1;
  background-color: #ffffff;
  padding: 11px 0;
  text-align: center;
  font: 14px/1.5 "Galano Grotesque Regular",sans-serif;
}

.c3:hover,
.c3:focus {
  background-color: #def5f4;
}

.c3:first-child {
  border-left: 2px solid #21bbb1;
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
}

.c3:last-child {
  border-top-right-radius: 8px;
  border-bottom-right-radius: 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<WithTheme(OptionGroup)
  name="deliveryDay"
  onSelect={[MockFunction]}
  options={
    Array [
      Object {
        "label": "Mon",
        "value": 0,
      },
      Object {
        "label": "Tue",
        "value": 1,
      },
      Object {
        "label": "Wed",
        "value": 2,
      },
      Object {
        "label": "Thu",
        "value": 3,
      },
    ]
  }
>
  <OptionGroup
    name="deliveryDay"
    onSelect={[MockFunction]}
    options={
      Array [
        Object {
          "label": "Mon",
          "value": 0,
        },
        Object {
          "label": "Tue",
          "value": 1,
        },
        Object {
          "label": "Wed",
          "value": 2,
        },
        Object {
          "label": "Thu",
          "value": 3,
        },
      ]
    }
    theme={
      Object {
        "borderRadius": Object {
          "normal": "8px",
          "small": "4px",
        },
        "brand": "dn",
        "colors": Object {
          "black": "#3a3a3a",
          "mint_green": "#47d7ac",
          "pepper_black_15": "#dadad9",
          "pepper_black_35": "#b1adab",
          "primary": "#21bbb1",
          "primary_15": "#def5f4",
          "primary_30": "#bceae7",
          "white": "#ffffff",
        },
        "fontFamilies": Object {
          "demi": "\\"Galano Grotesque Medium\\", sans-serif;",
          "medium": "\\"Galano Grotesque Medium\\", sans-serif;",
          "oldDemi": "\\"Dunbar Tall Medium\\", sans-serif;",
          "regular": "\\"Galano Grotesque Regular\\", sans-serif;",
        },
        "fontScale": Object {
          "body": "16px/1.5",
          "h1": "44px/1",
          "h2": "32px/1.3",
          "h3": "24px/1",
          "h4": "24px/1.5",
          "h5": "24px/0",
          "small": "14px/1.5",
          "tiny": "10px/1.5",
        },
        "fontStyle": Object {
          "body": "16px/1.5 \\"Galano Grotesque Regular\\", sans-serif;",
          "h1": "44px/1 \\"Galano Grotesque Medium\\", sans-serif;",
          "h2": "32px/1.3 \\"Galano Grotesque Medium\\", sans-serif;",
          "h3": "24px/1 \\"Galano Grotesque Medium\\", sans-serif;",
          "h4": "24px/1.5 \\"Galano Grotesque Regular\\", sans-serif;",
          "h5": "24px/0 \\"Galano Grotesque Medium\\", sans-serif;",
          "small": "14px/1.5 \\"Galano Grotesque Regular\\", sans-serif;",
          "tiny": "10px/1.5 \\"Galano Grotesque Medium\\", sans-serif;",
        },
        "iconSizes": Object {
          "large": "37px",
          "small": "22px",
        },
        "mealAttribute": Object {
          "color": "#3a3a3a",
          "margin": "0 10px 10px 0",
        },
        "mealTypeIcons": Object {
          "FISH": "/images/dn/fish.svg",
          "MEAT": "/images/dn/meat.svg",
          "VEGGIE": "/images/dn/veggie.svg",
        },
        "pill": Object {
          "borderRadius": "100px",
        },
        "spacings": Object {
          "large": "15px",
          "medium": "10px",
          "mega": "20px",
          "small": "5px",
        },
        "tonals": Object {
          "t10": "#747475",
          "t20": "#b9b9ba",
          "t30": "#f4f2f9",
        },
      }
    }
  >
    <styled.div>
      <div
        className="c0"
      >
        <OptionDn
          checked={true}
          key="0"
          label="Mon"
          name="deliveryDay"
          onChecked={[Function]}
          value={0}
        >
          <styled.label
            checked={true}
            htmlFor="deliveryDay__0"
            onChange={[Function]}
          >
            <label
              checked={true}
              className="c1"
              htmlFor="deliveryDay__0"
              onChange={[Function]}
            >
              <HiddenRadioButton
                id="deliveryDay__0"
                name="deliveryDay"
                value={0}
              >
                <styled.input
                  id="deliveryDay__0"
                  name="deliveryDay"
                  type="radio"
                  value={0}
                >
                  <input
                    className="c2"
                    id="deliveryDay__0"
                    name="deliveryDay"
                    type="radio"
                    value={0}
                  />
                </styled.input>
              </HiddenRadioButton>
              Mon
            </label>
          </styled.label>
        </OptionDn>
        <OptionDn
          checked={false}
          key="1"
          label="Tue"
          name="deliveryDay"
          onChecked={[Function]}
          value={1}
        >
          <styled.label
            checked={false}
            htmlFor="deliveryDay__1"
            onChange={[Function]}
          >
            <label
              checked={false}
              className="c3"
              htmlFor="deliveryDay__1"
              onChange={[Function]}
            >
              <HiddenRadioButton
                id="deliveryDay__1"
                name="deliveryDay"
                value={1}
              >
                <styled.input
                  id="deliveryDay__1"
                  name="deliveryDay"
                  type="radio"
                  value={1}
                >
                  <input
                    className="c2"
                    id="deliveryDay__1"
                    name="deliveryDay"
                    type="radio"
                    value={1}
                  />
                </styled.input>
              </HiddenRadioButton>
              Tue
            </label>
          </styled.label>
        </OptionDn>
        <OptionDn
          checked={false}
          key="2"
          label="Wed"
          name="deliveryDay"
          onChecked={[Function]}
          value={2}
        >
          <styled.label
            checked={false}
            htmlFor="deliveryDay__2"
            onChange={[Function]}
          >
            <label
              checked={false}
              className="c3"
              htmlFor="deliveryDay__2"
              onChange={[Function]}
            >
              <HiddenRadioButton
                id="deliveryDay__2"
                name="deliveryDay"
                value={2}
              >
                <styled.input
                  id="deliveryDay__2"
                  name="deliveryDay"
                  type="radio"
                  value={2}
                >
                  <input
                    className="c2"
                    id="deliveryDay__2"
                    name="deliveryDay"
                    type="radio"
                    value={2}
                  />
                </styled.input>
              </HiddenRadioButton>
              Wed
            </label>
          </styled.label>
        </OptionDn>
        <OptionDn
          checked={false}
          key="3"
          label="Thu"
          name="deliveryDay"
          onChecked={[Function]}
          value={3}
        >
          <styled.label
            checked={false}
            htmlFor="deliveryDay__3"
            onChange={[Function]}
          >
            <label
              checked={false}
              className="c3"
              htmlFor="deliveryDay__3"
              onChange={[Function]}
            >
              <HiddenRadioButton
                id="deliveryDay__3"
                name="deliveryDay"
                value={3}
              >
                <styled.input
                  id="deliveryDay__3"
                  name="deliveryDay"
                  type="radio"
                  value={3}
                >
                  <input
                    className="c2"
                    id="deliveryDay__3"
                    name="deliveryDay"
                    type="radio"
                    value={3}
                  />
                </styled.input>
              </HiddenRadioButton>
              Thu
            </label>
          </styled.label>
        </OptionDn>
      </div>
    </styled.div>
  </OptionGroup>
</WithTheme(OptionGroup)>
`;

exports[`OptionGroup renders correctly with ms theme 1`] = `
.c2 {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c1 {
  background-color: #ffffff;
  border: 1px solid #b1adab;
  border-right: none;
  border-top: 6px solid #dadad9;
  box-shadow: 0 -1px 0 #b1adab;
  cursor: pointer;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font: 14px/1.5 "Avenir Regular",sans-serif;
  margin: 0 0 25px;
  padding: 15px 0;
  text-align: center;
  border-top-color: #47d7ac;
  font-family: "Avenir Demi",sans-serif;
}

.c1:first-child {
  border-top-left-radius: 6px;
  border-bottom-left-radius: 6px;
}

.c1:last-child {
  border-right: 1px solid #b1adab;
  border-top-right-radius: 6px;
  border-bottom-right-radius: 6px;
}

.c3 {
  background-color: #ffffff;
  border: 1px solid #b1adab;
  border-right: none;
  border-top: 6px solid #dadad9;
  box-shadow: 0 -1px 0 #b1adab;
  cursor: pointer;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font: 14px/1.5 "Avenir Regular",sans-serif;
  margin: 0 0 25px;
  padding: 15px 0;
  text-align: center;
}

.c3:first-child {
  border-top-left-radius: 6px;
  border-bottom-left-radius: 6px;
}

.c3:last-child {
  border-right: 1px solid #b1adab;
  border-top-right-radius: 6px;
  border-bottom-right-radius: 6px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<WithTheme(OptionGroup)
  name="deliveryDay"
  onSelect={[MockFunction]}
  options={
    Array [
      Object {
        "label": "Mon",
        "value": 0,
      },
      Object {
        "label": "Tue",
        "value": 1,
      },
      Object {
        "label": "Wed",
        "value": 2,
      },
      Object {
        "label": "Thu",
        "value": 3,
      },
    ]
  }
>
  <OptionGroup
    name="deliveryDay"
    onSelect={[MockFunction]}
    options={
      Array [
        Object {
          "label": "Mon",
          "value": 0,
        },
        Object {
          "label": "Tue",
          "value": 1,
        },
        Object {
          "label": "Wed",
          "value": 2,
        },
        Object {
          "label": "Thu",
          "value": 3,
        },
      ]
    }
    theme={
      Object {
        "borderRadius": Object {
          "normal": "8px",
          "small": "4px",
        },
        "brand": "ms",
        "colors": Object {
          "black": "#2d2926",
          "mint_green": "#47d7ac",
          "pepper_black_15": "#dadad9",
          "pepper_black_35": "#b1adab",
          "primary": "#47d7ac",
          "white": "#ffffff",
        },
        "fontFamilies": Object {
          "demi": "\\"Avenir Demi\\", sans-serif;",
          "medium": "\\"Avenir Medium\\", sans-serif;",
          "regular": "\\"Avenir Regular\\", sans-serif;",
        },
        "fontScale": Object {
          "body": "16px/1.5",
          "h1": "36px/1.5",
          "h2": "32px/1.5",
          "h3": "28px/1",
          "h4": "24px/1.5",
          "h5": "20px/0",
          "small": "14px/1.5",
          "tiny": "10px/1.5",
        },
        "fontStyle": Object {
          "body": "16px/1.5 \\"Avenir Regular\\", sans-serif;",
          "h1": "36px/1.5 \\"Avenir Medium\\", sans-serif;",
          "h2": "32px/1.5 \\"Avenir Regular\\", sans-serif;",
          "h3": "28px/1 \\"Avenir Medium\\", sans-serif;",
          "h4": "24px/1.5 \\"Avenir Regular\\", sans-serif;",
          "h5": "20px/0 \\"Avenir Medium\\", sans-serif;",
          "small": "14px/1.5 \\"Avenir Regular\\", sans-serif;",
          "tiny": "10px/1.5 \\"Avenir Demi\\", sans-serif;",
        },
        "iconSizes": Object {
          "large": "37px",
          "small": "22px",
        },
        "mealTypeIcons": Object {
          "FISH": "/images/ms/fish.svg",
          "MEAT": "/images/ms/meat.svg",
          "VEGGIE": "/images/ms/veggie.svg",
        },
        "pill": Object {
          "borderRadius": "none",
        },
        "spacings": Object {
          "large": "15px",
          "medium": "10px",
          "mega": "15px",
          "small": "5px",
        },
        "tonals": Object {
          "t10": "#413b38",
          "t20": "#716d6a",
          "t30": "#f6f6f6",
        },
      }
    }
  >
    <styled.div>
      <div
        className="c0"
      >
        <Option
          checked={true}
          key="0"
          label="Mon"
          name="deliveryDay"
          onChecked={[Function]}
          value={0}
        >
          <styled.label
            checked={true}
            htmlFor="deliveryDay__0"
            onChange={[Function]}
          >
            <label
              checked={true}
              className="c1"
              htmlFor="deliveryDay__0"
              onChange={[Function]}
            >
              <HiddenRadioButton
                id="deliveryDay__0"
                name="deliveryDay"
                value={0}
              >
                <styled.input
                  id="deliveryDay__0"
                  name="deliveryDay"
                  type="radio"
                  value={0}
                >
                  <input
                    className="c2"
                    id="deliveryDay__0"
                    name="deliveryDay"
                    type="radio"
                    value={0}
                  />
                </styled.input>
              </HiddenRadioButton>
              Mon
            </label>
          </styled.label>
        </Option>
        <Option
          checked={false}
          key="1"
          label="Tue"
          name="deliveryDay"
          onChecked={[Function]}
          value={1}
        >
          <styled.label
            checked={false}
            htmlFor="deliveryDay__1"
            onChange={[Function]}
          >
            <label
              checked={false}
              className="c3"
              htmlFor="deliveryDay__1"
              onChange={[Function]}
            >
              <HiddenRadioButton
                id="deliveryDay__1"
                name="deliveryDay"
                value={1}
              >
                <styled.input
                  id="deliveryDay__1"
                  name="deliveryDay"
                  type="radio"
                  value={1}
                >
                  <input
                    className="c2"
                    id="deliveryDay__1"
                    name="deliveryDay"
                    type="radio"
                    value={1}
                  />
                </styled.input>
              </HiddenRadioButton>
              Tue
            </label>
          </styled.label>
        </Option>
        <Option
          checked={false}
          key="2"
          label="Wed"
          name="deliveryDay"
          onChecked={[Function]}
          value={2}
        >
          <styled.label
            checked={false}
            htmlFor="deliveryDay__2"
            onChange={[Function]}
          >
            <label
              checked={false}
              className="c3"
              htmlFor="deliveryDay__2"
              onChange={[Function]}
            >
              <HiddenRadioButton
                id="deliveryDay__2"
                name="deliveryDay"
                value={2}
              >
                <styled.input
                  id="deliveryDay__2"
                  name="deliveryDay"
                  type="radio"
                  value={2}
                >
                  <input
                    className="c2"
                    id="deliveryDay__2"
                    name="deliveryDay"
                    type="radio"
                    value={2}
                  />
                </styled.input>
              </HiddenRadioButton>
              Wed
            </label>
          </styled.label>
        </Option>
        <Option
          checked={false}
          key="3"
          label="Thu"
          name="deliveryDay"
          onChecked={[Function]}
          value={3}
        >
          <styled.label
            checked={false}
            htmlFor="deliveryDay__3"
            onChange={[Function]}
          >
            <label
              checked={false}
              className="c3"
              htmlFor="deliveryDay__3"
              onChange={[Function]}
            >
              <HiddenRadioButton
                id="deliveryDay__3"
                name="deliveryDay"
                value={3}
              >
                <styled.input
                  id="deliveryDay__3"
                  name="deliveryDay"
                  type="radio"
                  value={3}
                >
                  <input
                    className="c2"
                    id="deliveryDay__3"
                    name="deliveryDay"
                    type="radio"
                    value={3}
                  />
                </styled.input>
              </HiddenRadioButton>
              Thu
            </label>
          </styled.label>
        </Option>
      </div>
    </styled.div>
  </OptionGroup>
</WithTheme(OptionGroup)>
`;
